src/java.base/share/classes/java/lang/runtime/TemplateRuntime.java

```diff
@@ -0,0 +1,236 @@
+     *
+     * @throws NullPointerException if any of the arguments is null
+     * @throws Throwable            if linkage fails
+     */
+    public static CallSite processStringTemplate(MethodHandles.Lookup lookup,
+                                                 String name,
+                                                 MethodType type,
+                                                 MethodHandle processorGetter,
+                                                 String... fragments) throws Throwable {
+        Objects.requireNonNull(lookup, "lookup is null");
+        Objects.requireNonNull(name, "name is null");
+        Objects.requireNonNull(type, "type is null");
+        Objects.requireNonNull(processorGetter, "processorGetter is null");
+        Objects.requireNonNull(fragments, "fragments is null");
+
+        Processor<?, ?> processor = (Processor<?, ?>)processorGetter.invoke();
+        MethodHandle mh = processor instanceof Linkage linkage
+                ? linkage.linkage(List.of(fragments), type)
+                : defaultProcessMethodHandle(type, processor, List.of(fragments));
+
+        return new ConstantCallSite(mh);
+    }
+
+    /**
+     * Creates a simple {@link StringTemplate} and then invokes the processor's process method.
+     *
+     * @param fragments fragments from string template
+     * @param processor {@link Processor} to process
+     * @param values    array of expression values
+     *
+     * @return result of processing the string template
+     */
+    private static Object defaultProcess(
+            List<String> fragments,
+            Processor<?, ?> processor,
+            Object[] values
+    ) throws Throwable {
+        List<Object> asList = Collections.unmodifiableList(new ArrayList<>(Arrays.asList(values)));
+        return processor.process(StringTemplate.of(fragments, asList));
+    }
+
+    /**
+     * Generate a {@link MethodHandle} which is effectively invokes
+     * {@code processor.process(new StringTemplate(fragments, values...)}.
+     *
+     * @return default process {@link MethodHandle}
+     */
+    private static MethodHandle defaultProcessMethodHandle(
+            MethodType type,
+            Processor<?, ?> processor,
+            List<String> fragments
+    ) {
+        MethodHandle mh = MethodHandles.insertArguments(DEFAULT_PROCESS_MH, 0, fragments, processor);
+        return mh.asCollector(Object[].class, type.parameterCount()).asType(type);
+    }
+
+     /**
+     * Used to create a {@link StringTemplate} when number of value slots exceeds
+     * {@link java.lang.invoke.StringConcatFactory#MAX_INDY_CONCAT_ARG_SLOTS}.
+     *
+     * @param fragments  array of string fragments
+     * @param values     array of values
+     *
+     * @return new {@link StringTemplate}
+     */
+    private static StringTemplate fromArrays(String[] fragments, Object[] values) {
+        return StringTemplateImplFactory.newStringTemplate(fragments, values);
```

Comment:
Should perform defensive copying for the input arrays, specifically `values` (fragments is passed to `List.of`, which is already safe)

In addition, I recommend renaming methods that don't perform defensive copies to like `ofTrusted`, so we can better avoid such problems.